flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -1217,40 +1228,31 @@ public void onDetachedFromActivity() {
   }
 
   @Override
-  public void onMethodCall(MethodCall call, Result result) {
+  public void onMethodCall(MethodCall call, @NonNull final Result result) {
     switch (call.method) {
       case INITIALIZE_METHOD:
-        {
-          initialize(call, result);
-          break;
-        }
+        initialize(call, result);
+        break;
       case GET_NOTIFICATION_APP_LAUNCH_DETAILS_METHOD:
-        {
-          getNotificationAppLaunchDetails(result);
-          break;
-        }
+        getNotificationAppLaunchDetails(result);
+        break;
       case SHOW_METHOD:
-        {
-          show(call, result);
-          break;
-        }
+        show(call, result);
+        break;
       case SCHEDULE_METHOD:
-        {
-          schedule(call, result);
-          break;
-        }
+        schedule(call, result);
+        break;
       case ZONED_SCHEDULE_METHOD:
-        {
-          zonedSchedule(call, result);
-          break;
-        }
+        zonedSchedule(call, result);
+        break;
+      case REQUEST_PERMISSION_METHOD:
+        requestPermission(result::success);
```

Comment:
I've saw that multiple calls to this method can result in a crash. To reproduce this, make two consecutive calls from the Flutter side and tap on "Allow" to grant permissions. App will crash and can see logs on requesting multiple permissions, reply already received. This can be avoided by limiting the plugin to only handle one permissions request at a time and return an error back to consumers if there are concurrent calls. Examples I've seen on how to handle this can be found in plugins like `google_sign_in` (see https://github.com/flutter/plugins/blob/e8a7abcd654f53d9111cb273dd24a7a7c5974b7a/packages/google_sign_in/google_sign_in_android/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java#L284, I used this approach in `flutter_appauth`) and `permission_handler` (see https://github.com/Baseflow/flutter-permission-handler/blob/051828f0c5b572fde65ef3474af0de29407fe9aa/permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java#L210)

Reply:
Thanks, I'll look into this during the weekend.

Reply:
It's probably becuase you were busy that this hasn't been addressed yet but in case it's because you forgot, a reminder that this issue remains

Reply:
Yes, I want to do this today. Thank you :)

Reply:
Done.